.travis.yml

```diff
@@ -1,46 +1,43 @@
 dist: bionic
 sudo: required
 language: minimal
-addons:
-  apt:
-    sources:
-    - sourceline: "ppa:chris-lea/redis-server"
-    packages:
-    - redis-server
-    - redis-tools
 
 branches:
   only:
   - master
 
 services:
 - docker
-- redis-server
 
-before_install:
-- sudo apt --yes --quiet update
+cache:
+  directories:
+  - docker_images
 
-# Use PostgreSQL 11
-- sudo apt --yes --quiet autoremove --purge libpq\* postgresql-{,client-,contrib-}{9,10}\*
-- sudo apt --yes --quiet install postgresql-11
+before_install:
+# Load saved Docker images from Travis' cache
+- gunzip --stdout docker_images/images.tar.gz | docker load || true
 
 install:
-- docker build --build-arg build_type=dev --tag azafea-tests .
-
-before_script:
-# Create the PostgreSQL role and database
-- sudo -u postgres createuser azafea
-- sudo -u postgres createdb --owner=azafea azafea-tests
+- docker build --build-arg build_type=dev --tag azafea-tests --pull .
 
-# Set the default passwords for PostgreSQL and Redis, as that's what the tests expect
-- sudo -u postgres psql --command="ALTER USER azafea WITH PASSWORD 'CHANGE ME!!';"
-- redis-cli config set requirepass 'CHANGE ME!!'
+before_cache:
+# Save Docker images to Travis' cache, but prune old ones first
+- mkdir -p docker_images
+- docker images --quiet --filter=dangling=true | xargs -r docker rmi
+- docker save $(docker images -a -q) | gzip > docker_images/images.tar.gz
 
 matrix:
   include:
   - name: "Integration Tests"
+    before_script:
+    - docker image pull postgres:11-alpine
+    - docker image pull redis:5-alpine
```

Comment:
You use these ones because they are smaller? (that's fine, I just want to make sure I understand the "why")

Reply:
Just for the smaller images (faster to download). There are actually tags for the latest Alpine variants. But this could be surprising (builds could suddenly fail if the image in Docker Hub is updated to a new major version).

Reply:
> There are actually tags for the latest Alpine variants

Ah, they are caled "alpine"! I missed them because I was searching for something with "latest" in the name.

> But this could be surprising (builds could suddenly fail if the image in Docker Hub is updated to a new major version).

In general I agree with that.

For things like Redis and PostgreSQL though, I'm pretty confident about their consistent stability and compatibility that I'd be willing to risk it.

Although I guess it's nice to test with what is in production right now as well… :thinking: 

I know we use PostgreSQL 11 in production, are we also using Redis 5?

Reply:
We're on Redis 5.0. Since for now we're the only users of this we can set it to what Endless has in production. But for a more broad appeal we can actually run against multiple versions of PG and/or Redis (whatever we decide is supported). Personally I prefer to be explicit in this regard. I imagine that when we decide to update PG or Redis in production we'll start by adding a job to test on both versions (current and future) and keep that until all environments are updated to the new version and then we'll remove the old version from the CI.

Reply:
> Since for now we're the only users of this we can set it to what Endless has in production.

Yes, let's.